platform/macos/display_server_macos.mm

```diff
@@ -2290,6 +2290,32 @@
 	wd.mpath = p_region;
 }
 
+bool DisplayServerMacOS::window_is_centered(WindowID p_window) {
+	_THREAD_SAFE_METHOD_
+
+	if (!windows.has(p_window)) {
+		return false;
+	}
+
+	Point2i screen_size = screen_get_size(window_get_current_screen(p_window));
+	Point2i window_size = window_get_size(p_window);
+	Point2i position = screen_get_position(window_get_current_screen(p_window)) + screen_size / 2 - window_size / 2;
+
+	return (window_get_position(p_window) == position);
```

Comment:
This won't reliably work, the result might be (and almost always is) off by one pixel due to rounding errors.

Reply:
This might be a result of macOS hiDPI scaling.